### PR TITLE
chore(Jest): algoliasearch.helper [PART-4]

### PR DESCRIPTION
**Pull request by samouss**
This PR moves the folder `test/spec/algoliasearch.helper` to Jest. You can run the test locally with:

```sh
yarn jest test/spec/algoliasearch.helper
```

The migration has been done by [`jest-codemods`](https://github.com/skovhus/jest-codemods) (recommended by the [Jest](https://jestjs.io/docs/en/migration-guide#jest-codemods) project) for all the assertions. Then I removed the dependency on `sinon` file by file. I tried to didn't update the code style or anything unless it was too complicated to migrate.

### CODE REVIEW

File: test/spec/algoliasearch.helper/clears.js [hunk 1]
```diff
@@ -29,88 +28,74 @@ function fixture() {
...
+  expect(helper.state.numericRefinements.facet1).toEqual({'>=': [10]});
+  expect(helper.state.facetsExcludes.facet1).toEqual(['value']);
 
   helper.clearRefinements('facet1');
-  t.assert(isUndefined(helper.state.facetsRefinements.facet1));
-  t.assert(isUndefined(helper.state.numericRefinements.facet1));
-  t.assert(isUndefined(helper.state.facetsExcludes.facet1));
-
-  t.end();
+  expect(isUndefined(helper.state.facetsRefinements.facet1)).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/clears.js]: The assertion can be stricter.

File: test/spec/algoliasearch.helper/client.js [hunk 2]
```diff
@@ -1,113 +1,102 @@
...
 
-test('addAlgoliaAgent gets called if exists', function(t) {
+test('addAlgoliaAgent gets called if exists', function() {
   var client = {
-    addAlgoliaAgent: sinon.stub()
+    addAlgoliaAgent: jest.fn()
   };
 
-  t.notOk(client.addAlgoliaAgent.called);
+  expect(client.addAlgoliaAgent).not.toHaveBeenCalled();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/client.js]: We can use `toHaveBeenCalledTimes(0)` for better error messages and easier refactors.

File: test/spec/algoliasearch.helper/client.js [hunk 3]
```diff
@@ -1,113 +1,102 @@
...
 
   algoliaSearchHelper(client);
 
-  t.ok(client.addAlgoliaAgent.called);
-
-  t.end();
+  expect(client.addAlgoliaAgent).toHaveBeenCalled();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/client.js]: We should use `toHaveBeenCalledTimes(n)`.

File: test/spec/algoliasearch.helper/derive/detach.js [hunk 4]
```diff
@@ -20,34 +18,14 @@ test('[derived helper] detach a derived helper', function(t) {
...
-        'the helper generates the same query twice'
-      );
-      t.equal(
-        derivedHelper.listeners('result').length,
-        1,
-        'one listener is plugged to the derived helper'
-      );
+      expect(requests.length).toBe(2);
+      expect(requests[0]).toEqual(requests[1]);
+      expect(derivedHelper.listeners('result').length).toBe(1);
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/derive/detach.js]: We can use [`toHaveLength(n)`](https://jestjs.io/docs/en/expect#tohavelengthnumber).

File: test/spec/algoliasearch.helper/distinct.js [hunk 5]
```diff
@@ -1,69 +1,66 @@
...
+test('Distinct not set', function() {
   var helper = algoliasearchHelper(fakeClient, null, {
     facets: ['facetConj'],
     disjunctiveFacets: ['facet']
   });
   var state0 = helper.state;
 
   var disjunctiveFacetSearchParam = requestBuilder._getDisjunctiveFacetSearchParams(helper.state);
-  t.equal(disjunctiveFacetSearchParam.distinct, undefined, '[disjunctive] distinct should be undefined');
+  expect(disjunctiveFacetSearchParam.distinct).toBe(undefined);
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/distinct.js]: We can use `toBeUndefined()`.

File: test/spec/algoliasearch.helper/distinct.js [hunk 6]
```diff
@@ -1,69 +1,66 @@
...
   helper.setQuery('not empty');
   disjunctiveFacetSearchParam = requestBuilder._getDisjunctiveFacetSearchParams(helper.state);
-  t.equal(disjunctiveFacetSearchParam.distinct, undefined, '[disjunctive][query not empty] distinct should be undefined');
+  expect(disjunctiveFacetSearchParam.distinct).toBe(undefined);
   facetSearchParam = requestBuilder._getHitsSearchParams(helper.state);
-  t.equal(facetSearchParam.distinct, undefined, '[hits][query not empty] distinct should be undefined');
+  expect(facetSearchParam.distinct).toBe(undefined);
   forEach(requestBuilder._getQueries('', helper.state), function(q) {
-    t.notOk(q.hasOwnProperty('distinct'), '[hits][query not empty] no distinct should be in the queries by default');
+    expect(q.hasOwnProperty('distinct')).toBeFalsy();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/distinct.js]: We can be more strict.

File: test/spec/algoliasearch.helper/getQueryParameter.js [hunk 7]
```diff
@@ -14,11 +13,9 @@ test('getQueryParameter', function(t) {
...
-  t.equal(helper.getQueryParameter('minWordSizefor1Typo'), 8);
-  t.equal(helper.getQueryParameter('ignorePlurals'), true);
+  expect(helper.getQueryParameter('facets')).toEqual(['facet1']);
+  expect(helper.getQueryParameter('minWordSizefor1Typo')).toBe(8);
+  expect(helper.getQueryParameter('ignorePlurals')).toBe(true);
 
-  t.throws(bind(helper.getQueryParameter, helper, 'unknown'));
-
-  t.end();
+  expect(bind(helper.getQueryParameter, helper, 'unknown')).toThrow();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/getQueryParameter.js]: We can add an inline snapshot.

File: test/spec/algoliasearch.helper/incorrectFacetDefinition.js [hunk 8]
```diff
@@ -1,38 +1,31 @@
...
 
-var test = require('tape');
 var algoliasearchHelper = require('../../../index');
 
 var _ = require('lodash');
 
 var fakeClient = {};
 
-test('Conjuctive facet should be declared to be refined', function(t) {
+test('Conjuctive facet should be declared to be refined', function() {
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/incorrectFacetDefinition.js]:
```suggestion
test('Conjunctive facet should be declared to be refined', function() {
```

File: test/spec/algoliasearch.helper/incorrectFacetDefinition.js [hunk 9]
```diff
@@ -1,38 +1,31 @@
...
   var h = algoliasearchHelper(fakeClient, '', {});
 
-  t.throws(_.bind(h.addRefine, h, 'undeclaredFacet', 'value'), 'Adding a facet refinement should not be possible');
-  t.throws(_.bind(h.removeRefine, h, 'undeclaredFacet', 'value'), 'Remove a facet refinement should not be possible');
-  t.throws(_.bind(h.isRefined, h, 'undeclaredFacet', 'value'), 'Checking if a facet is refined should not be possible');
-
-  t.end();
+  expect(_.bind(h.addRefine, h, 'undeclaredFacet', 'value')).toThrow();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/incorrectFacetDefinition.js]: Do we want to track which errors are thrown?

File: test/spec/algoliasearch.helper/numericFilters.js [hunk 10]
```diff
@@ -73,113 +72,80 @@ test('Numeric filters: numeric filters from setters', function(t) {
...
+    expect(helper.getRefinements('attribute')).toEqual([]);
   }
 );
 
-test('Should be able to get if an attribute has numeric filter with hasRefinements', function(t) {
+test('Should be able to get if an attribute has numeric filter with hasRefinements', function() {
   var helper = algoliasearchHelper(fakeClient, null, null);
 
-  t.notOk(helper.hasRefinements('attribute'), 'not refined initially');
+  expect(helper.hasRefinements('attribute')).toBeFalsy();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/numericFilters.js]: Can be more strict.

File: test/spec/algoliasearch.helper/pages.js [hunk 11]
```diff
@@ -1,72 +1,68 @@
...
+var bind = require('lodash/bind');
+
 var fakeClient = {};
 
-test('setChange should change the current page', function(t) {
+test('setChange should change the current page', function() {
   var helper = algoliasearchHelper(fakeClient, null, null);
 
-  t.ok(helper.getCurrentPage() === 0, 'First page should be 0');
+  expect(helper.getCurrentPage() === 0).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/algoliasearch.helper/pages.js]: We can use `toBe(0)`.

File: test/spec/algoliasearch.helper/clears.js (hunk 1 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/clears.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/client.js (hunk 2 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/client.js]: Nope we can't rely on the number of call for this test. It is actually called multiple times but it's an implementation detail that we don't care about (the client deduplicate them). We could use `toHaveBeenCalledTimes` for the first one but not the second. To be consistent for the two assertions I've used `toHaveBeenCalled` on both.

File: test/spec/algoliasearch.helper/client.js (hunk 3 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/client.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/derive/detach.js (hunk 4 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/derive/detach.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/distinct.js (hunk 5 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/distinct.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/distinct.js (hunk 6 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/distinct.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/getQueryParameter.js (hunk 7 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/getQueryParameter.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278627422

File: test/spec/algoliasearch.helper/incorrectFacetDefinition.js (hunk 9 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/incorrectFacetDefinition.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/numericFilters.js (hunk 10 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/numericFilters.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/algoliasearch.helper/pages.js (hunk 11 above)
Comment (samouss) [reply test/spec/algoliasearch.helper/pages.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

**Pull request merged**